lib/matplotlib/lines.py

```diff
@@ -89,24 +89,24 @@ def segment_hits(cx, cy, x, y, radius):
     # Only look at line segments whose nearest point to C on the line
     # lies within the segment.
     dx, dy = x[1:] - xr, y[1:] - yr
-    Lnorm_sq = dx ** 2 + dy ** 2  # Possibly want to eliminate Lnorm==0
-    u = ((cx - xr) * dx + (cy - yr) * dy) / Lnorm_sq
-    candidates = (u >= 0) & (u <= 1)
+    u = (cx - xr) * dx + (cy - yr) * dy
+    candidates = (u >= 0) & (u <= dx ** 2 + dy ** 2)
```

Comment:
Can you change this to:
```suggestion
    candidates = (0 <= u) & (u <= dx ** 2 + dy ** 2)
```
it's a little nicer.

Reply:
agreed